### PR TITLE
Update ve theory

### PR DESCRIPTION
**Pull request by vgro**
This PR adds some general info to the theory of the VR, including Figure 1 from the overview paper and the user stories. I'm not sure if this is the best way to start this section, comments are welcome.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [ ] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works
- [x] Relevant documentation reviewed and updated

### CODE REVIEW

File: docs/source/virtual_ecosystem/theory.md [hunk 1]
```diff
@@ -1 +1,79 @@
 # The theory of the Virtual Ecosystem
+
+Ecosystems are complex systems that arise from the interplay between animals, plants,
+and microbes with their abiotic environment (Figure 1). Many of these interactions are
```
Comment (jacobcook1995) [inline docs/source/virtual_ecosystem/theory.md:4]: "within"? (though must say I'm not 100% sure of the proper grammar)

**Review by jacobcook1995 — APPROVED** [review]
Content looks really great to me!

**Comment by vgro** [comment]
I think the user stories could probably get their own page?

**Review by TaranRallings — APPROVED** [review]
I think this looks great. It's concise while covering all of the core material.

**Comment by vgro** [comment]
@davidorme  could you provide the figure please?

**Comment by davidorme** [comment]
@vgro Figure added!

**Review by davidorme — COMMENTED** [review]
LGTM.

**Pull request merged**